### ISSUE TITLE
Establish a consistent pattern for navigation buttons that open sub-panels

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66448]
### Description

In the Site editor, some settings panels have sub-panels (drill-down menus in design jargon). The controls to open these sub-panels are usually `NavigationButtonAsItem` components (under the hod it's a `Navigator.Button`).

Some of these buttons do show a chevron icon to visually communicate they open a drill-down menu. Some don't.

It's not that clear to me what the criteria is to determine which buttons should show a chevron and which ones should not.

There may be reasons why some don't but I'd tend to think a chevron icon adds value as it makes the user experiences more predictable and communicates well what's going to happen when clicking one of these buttons.

There are other minor inconsistencies, for example some of these buttons have a border. Some don't.
I do realize the navigation buttons that are grouped in a set of logically related items need to be styled a bit differently. Still, the lack of a chevron icon introduces inconsistency and doesn't help predictability.

A few examples:

The following navigation buttons show a chevron icon:
- Styles > Colors > Edit palette 
- Styles > Typography > Font size presets 
- Styles > Browse styles 
- Styles > Blocks 
- Styles > Additional CSS

![Image](https://github.com/user-attachments/assets/dedacff7-2088-4b6a-bf90-aa701b67e503)

The following navigation buttons do not show a chevron icon:
- Styles root menu: Typography, Colors, Background, Shadows, Layout.
- Styles > Typography > Elements

![Image](https://github.com/user-attachments/assets/630d8f41-24a8-4e12-b33b-28d348e12497)

The buttons undeer 'Elements' are particularly confusing to me. They look like the fonts ones above. However, the fonts ones open a modal dialog while  the Elements ones open a drill-down menu. They basically look the same, but they do different things.

One more example:
- The custom shadows navigation buttons don't show a chevron icon. It's particularly confusing when there's only _one_ custom shadow:

![Image](https://github.com/user-attachments/assets/93ab8cac-942d-4c18-aaaa-a0cd55335e82)

I'd like to propose to use a chevron icon for _all_ the buttons that open a drill-down menu. 
Also i'd suggest to consider some more consistency about borders and margins.

### Step-by-step reproduction instructions

N/A
See the examples in the Site editor > Styles panel 

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by afercia** [comment]
Re: borders around the items: quoting @jameskoster from https://github.com/WordPress/gutenberg/pull/66476#issuecomment-2541268902

> I lean towards no strokes, because the 'contained' style feels better suited when items in the list are directly related to one another, e.g. managing shadow styles or font size presets.

I would agree that borders should either be used with some consistent meaning ot not be used at all.

What about:
- Use borders to separate items that are _not logically grouped_ e.g. the items in the root of the menu: Typography, Colors, background...
- Do not use borders, and maybe use a border around the group, for items that are logically grouped.

**Comment by jameskoster** [comment]
I agree it would be beneficial to tidy this up. There's a lot to cover, so when we arrive at a satisfactory design we might start by documenting this on Storybook as a source of truth before implementation.

> Use borders to separate items that are not logically grouped e.g. the items in the root of the menu: Typography, Colors, background...

The two 'types' of groups identified make sense, but in this case I would just use no borders. It seems to me the expectation is that menu items are unrelated unless otherwise indicated by the UI.

For instance _adding_ a stroke to the container helps indicate the relationship of the children, and is a common existing pattern.

<img width="1013" alt="Image" src="https://github.com/user-attachments/assets/fc3133fd-ed7c-4bfd-a0c4-f091738a7e01" />

Other considerations:

* Should we use the smaller chevron icon (like in the navigation sidebar)? This might help indicate at the decorative nature of this element.
* Should the chevron only appear on hover/focus to avoid adding unnecessary noise to the UI?

**Comment by afercia** [comment]
> For instance adding a stroke to the container helps indicate the relationship of the children

I'd agree a border _around_ items visually suggests they're logically grouped items.
A order _between_ items should be used to visually communicate they're logically separated.

**Comment by jameskoster** [comment]
I don't think we need to explicitly indicate both cases. If a border on the container denotes the relationship of the children, then the _absence_ of a border inherently indicates the opposite, with clearer visual distinction.

Let's see what @WordPress/gutenberg-design thinks though.

**Comment by jasmussen** [comment]
Chevron to indicate drilldown makes sense, and related items in an itemgroup (bordered) could similarly have chevrons. In the case of fonts, though, these open a modal, so they should probably not have the chevron.

**Comment by jameskoster** [comment]
@jasmussen we're discussing patterns that indicate when drilldown items are directly related (e.g. font size presets) versus those that are not (e.g. top level menu items). Examples in https://github.com/WordPress/gutenberg/issues/66448#issuecomment-2541508500.

Also:

> * Should we use the smaller chevron icon (like in the navigation sidebar)? This might help indicate at the decorative nature of this element.
> * Should the chevron only appear on hover/focus to avoid adding unnecessary noise to the UI?

**Comment by auareyou** [comment]
It seems the immediate accessible solution here would be to add chevrons (or another clear affordance) that make the drill-down behavior obvious to everyone. However, if adding chevrons to all buttons introduces too much visual noise—which I agree it does—it might be worth reconsidering the overall visual language around settings.

For instance, if we take the Typography roles list as an example, perhaps changes could happen in place with a different type of control, such as a select menu, an accordion, or another component that doesn’t pull users away from the main view of the typography panel.

Looking at industry standards in other design and development tools, the inspector panel typically keeps sections corresponding to the user’s selection always visible. Advanced settings or drill-downs often occur within popovers, reducing the number of "round trips" while still allowing for focused interactions.

Example from Figma:
<img width="559" alt="Image" src="https://github.com/user-attachments/assets/9a3eb158-566b-45f9-8605-1b9e548240a1" />

Example from Webflow:
<img width="246" alt="Image" src="https://github.com/user-attachments/assets/76e25963-185c-4e22-97f6-961ec517035a" />

**Comment by jasmussen** [comment]
Nice one. Yep, the flyouts are useful, and often used in conjunction with the [ItemGroup](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-itemgroup--docs). Some of the challenge lies in the cases where a drilldown/sub-panel is already in use: what's the best pattern for those? 

Absent larger undertakings, showing the drilldown on hover seems a good immediate path forward for those.

**Comment by matt-west** [comment]
+1 to adding chevrons to indicate a drill-down. Making these visible on hover/focus makes sense to avoid unnecessary visual noise.

Do we need the border to visually group related items? There’s usually a header that communicates the items are related.

Examples:

<img width="270" alt="Image" src="https://github.com/user-attachments/assets/ed934aa8-18fa-45f5-a53b-c7e1b650c6c5" />
<img width="270" alt="Image" src="https://github.com/user-attachments/assets/fbe5802e-59f2-4be3-b400-e316511466a2" />
<img width="269" alt="Image" src="https://github.com/user-attachments/assets/b96171b4-1272-4093-a27d-41bdb55cc252" />

**Comment by afercia** [comment]
> Making these visible on hover/focus makes sense 

I would recommend against making the chevrons only visible on hover/focus. Some users do use devices and technologies that don't use hover/focus. Voice control or speech recognition software, for example. Cognitive load is also impacted with information that is only shown on hover/focus. Providing an equivalent experience to all users is a principle of inclusivity and accessibility and it should prevail on the argument of making the UI 'cleaner', which is highly subjective.

**Comment by fcoveram** [comment]
+1 to show the chevron icon to indicate drill-down. And I second @afercia suggestion of not hiding it on hover/focus.

**Comment by jarekmorawski** [comment]
I agree that we should show the chevron icon at all times, not just on hover/focus. Otherwise, it may not be clear to some users that they're drilldowns; it's like making a door handle magically appear only when you're about to leave the room.

The icon will also help differentiate drilldowns from buttons that open popovers (like the background image button in the Group block in the editor), which could become a problem should we decide to consolidate the appearance of such elements across the whole UI.

**Comment by afercia** [comment]
Related to the ItemGroup styling and the discussion about borders to indicate grouping / separation, it's worth reminding that other components don't use the ItemGroup directly but they do _emulate_ its styling. For example, the color settings:

![Image](https://github.com/user-attachments/assets/40c6ee4d-8c6f-41d6-9546-d71464d1e0a5)

Related to the ongoing discussion on https://github.com/WordPress/gutenberg/issues/58936

**Comment by afercia** [comment]
Also relevant: https://github.com/WordPress/gutenberg/issues/55730

In teh Inserter, navigational items that open sub-panels now show the chevron icon only on hover/focus _and_ as indication of the selected state. That's one more inconsistency.

**Comment by jameskoster** [comment]
> Cognitive load is also impacted with information that is only shown on hover/focus

I think it can be argued that this goes both ways. Once you're familiar with the behviours of the UI, certain visual cues risk becoming unwelcome distractions. 

I'm not against making the chevrons permanently visible, but I wonder if we can reduce their relative prominence a little, in an accessible way. For instance using the `small` chevron icon variant (like we do in the dark sidebar), or a lighter color, or both.

<img width="1013" alt="Image" src="https://github.com/user-attachments/assets/7e3afcef-c99e-49d5-b41b-8605c6be8136" />

I'm also curious about how this should be managed in a sustainable way. Do we need a new component?

**Comment by afercia** [comment]
I wouldn't be opposed to try a smaller icon. The current one looks big indeed. Can we use an icon where the actual glyph is 24 by 24 pixels? It's not a strict requirement, I'm just borrowing it from the WCAG target size, but I can't think to any other reference to determine how bic an icon should be to be perceived by everyone regardless of ability.

**Comment by jameskoster** [comment]
I fear a 24px glyph would look comically large and entirely out of proportion with the rest of the UI 🙈

<img width="303" alt="Image" src="https://github.com/user-attachments/assets/600f6457-3750-4406-9356-c1af765aa53e" />

**Comment by afercia** [comment]
> I fear a 24px glyph would look comically large and entirely out of proportion with the rest of the UI 🙈

Ouch, yes, I see 😿  Hm then I would say the icon _glyph_ should be no smaller than the uppercase font?

**Comment by afercia** [comment]
One more inconsistency is for the list of blocks in Styles > Blocks. They do open a drill-down sup-panel but there's no visual indication at all that they are navigational items:

![Image](https://github.com/user-attachments/assets/1d26010f-059c-4bf0-8925-a5163a6b9344)

**Comment by jameskoster** [comment]
Extracting from the conversation in https://github.com/WordPress/gutenberg/issues/55730; Let's talk about [vertical tabs](https://wordpress.github.io/gutenberg/?path=/story/components-tabs--vertical). Specifically, should tabs be treated the same way as drilldowns whereby the chevron is permanently visible?

### Arguments for showing the chevron permanently

Vertical tabs function similarly to drilldowns (e.g., loading or navigating to a new panel or view), so using the same icon could establish a consistent expectation for how the interaction works.

When users encounter tabs and drilldowns within the same interface, consistency in icon usage can reduce cognitive load. For example, they might associate the chevron with "navigating into related content" universally.

### Arguments against

Drilldowns represent hierarchical navigation where selecting an option moves the user deeper into a structure, potentially hiding the current view.

Meanwhile, tabs typically represent parallel and equally accessible content sections. A user clicks a tab, and content changes in the same view without implying hierarchy or navigation depth. Adding a chevron to the tabs might incorrectly suggest hierarchical navigation or a drilldown behavior, which is not the case.

Icons are an important visual cue but should match the underlying behavior. If the chevron is used inconsistently or inappropriately, it could lead to confusion for users relying on visual scanning or assistive technologies that describe icons.

---

On balance I think I would prefer to treat tabs differently to drilldowns; only show the icon on hover/focus/active states. I'd even consider using a different shape to avoid confusion with drilldowns. There could be opportunity for closer alignment between horizontal / vertical tab treatment. Rough sketch:

<img width="866" alt="Image" src="https://github.com/user-attachments/assets/d760d00f-578c-41df-94a9-111fd864c785" />

**Comment by afercia** [comment]
Thanks for your thorough analysis @jameskoster. To me, the underlying problem is that the current workings of the vertical tabs with no tabpanel open by default _resambles_ the interaction model of a drilldown menu. Of course there are differences as you pointed out. 

> consistent expectation for how the interaction works

In the above scenario, the interaction works similarly to a drilldown menu. As such, I would say the expectation should be made consistent by providing a similar visual indication.

However, the vertical tabs with no tabpanel open by default is fundamentally a broken pattern. It shouldn't be used that way. Tabs should have a tabpanel open by default. In this case, a chevron icon would be inappropriate. Rather, the design itself of the tabs/tabpanels should make clear the expected interaction. I would argue that the current 'light' design of the tabs could be reconsidered because they don't look like tabs. [Please excuse the crudity of the screenshot below](https://www.youtube.com/watch?v=Yxx5yjWlxQo), it's just to illustrate how the design itself of the active tab and the active tabpanel should make clear what the expected interaction is:

![Image](https://github.com/user-attachments/assets/c928e32f-60c1-44ef-8132-a7073ba1f659)

**Comment by jameskoster** [comment]
> that the current workings of the vertical tabs with no tabpanel open by default

For the purpose of this discussion I think we should put that to one side. It is a separate issue relating specifically to the Inserter. When making decisions about the _components_ we should assume appropriate use. In other words, assuming a panel is visible by default, how does that change your thinking on this, do you still feel the chevrons should be permanently visible?

**Comment by afercia** [comment]
Well, assuming the Tabs component is used appropriately with a tabpanel open by default, no the chevron wouldn't be appropriate. However, I'd still love to see improvements to the Tabs component design to make the expected interaction clearer.

**Comment by afercia** [comment]
To illustrate: this is an example from the current Tabs in the preferences dialog. The chevron for hover/focus/selected doesn't make this UI look like a tabbed interface and I wouldn't expect the Tabs interaction model to work here. As you pointed out, a good UI should establish a consistent expectation for how the interaction works. Which doesn't appear to be the case here:

![Image](https://github.com/user-attachments/assets/faa74932-0bec-4a1b-915b-1cb3a19bac99)

**Comment by jameskoster** [comment]
Thanks for clarifying. So in terms of this issue we don't need to change anything about the Tabs component. Separately we might consider: 

1. Updating `Tabs` behavior in the Inserter to ensure a tab is selected by default, or use a different component.
2. Update Tab visual design (more cohesion between vertical/horizontal, more 'tab-like' appearance).

Feel free to open new issues to discuss those details.

---

Going back to the drilldowns, unless I'm mistaken there seems to be consensus around; 

* Show chevrons permanently.
* Use the `chevronRightSmall` icon.
* _Try_ a lighter (but still accessible) color ($gray-700).

Want to update the issue/labels to make this actionable, or did I miss something?

**Comment by afercia** [comment]
> Separately we might consider:
> 
> 1. Updating `Tabs` behavior in the Inserter to ensure a tab is selected by default, or use a different component.
> 2. Update Tab visual design (more cohesion between vertical/horizontal, more 'tab-like' appearance).

Yes I'd agree those should be addressed separately. Thanks for summarizing the action points @jameskoster.
I'd appreciate follow-up issues to be created and prioritised with some decently high priority as the Tabs UI is a pretty important UI in the editor.
